tests/ghost_test.go

```diff
@@ -0,0 +1,196 @@
+	}
+	parser := ghostsql.NewParserFromAlterStatement(migrationContext.AlterStatement)
+	migrationContext.AlterStatementOptions = parser.GetAlterStatementOptions()
+
+	if migrationContext.DatabaseName == "" {
+		if parser.HasExplicitSchema() {
+			migrationContext.DatabaseName = parser.GetExplicitSchema()
+		} else {
+			return nil, fmt.Errorf("database must be provided and database name must not be empty, or alterStatement must specify database name")
+		}
+	}
+	if migrationContext.OriginalTableName == "" {
+		if parser.HasExplicitTable() {
+			migrationContext.OriginalTableName = parser.GetExplicitTable()
+		} else {
+			return nil, fmt.Errorf("table must be provided and table name must not be empty, or alterStatement must specify table name")
+		}
+	}
+	migrationContext.ServeSocketFile = fmt.Sprintf("/tmp/gh-ost.%s.%s.sock", migrationContext.DatabaseName, migrationContext.OriginalTableName)
+	migrationContext.SetHeartbeatIntervalMilliseconds(heartbeatIntervalMilliseconds)
+	migrationContext.SetNiceRatio(niceRatio)
+	migrationContext.SetChunkSize(chunkSize)
+	migrationContext.SetDMLBatchSize(dmlBatchSize)
+	migrationContext.SetMaxLagMillisecondsThrottleThreshold(maxLagMillisecondsThrottleThreshold)
+	migrationContext.SetDefaultNumRetries(defaultNumRetries)
+	migrationContext.ApplyCredentials()
+	if err := migrationContext.SetCutOverLockTimeoutSeconds(cutoverLockTimoutSeconds); err != nil {
+		return nil, err
+	}
+	if err := migrationContext.SetExponentialBackoffMaxInterval(exponentialBackoffMaxInterval); err != nil {
+		return nil, err
+	}
+	return migrationContext, nil
+}
+
+func TestGhostSimpleNoop(t *testing.T) {
+	const port = 13306
+	basedir := t.TempDir()
+	datadir := filepath.Join(basedir, "data")
+	if err := os.Mkdir(datadir, 0755); err != nil {
+		t.Fatal(err)
+	}
+	mysql, err := mysql.Install(basedir, datadir, "root")
+	if err != nil {
+		t.Fatalf("failed to start MySQL: %v", err)
+	}
+	if err := mysql.Start(port, os.Stdout, os.Stderr, 60); err != nil {
+		t.Fatalf("failed to start MySQL: %v", err)
+	}
+
+	defer func() {
+		err := mysql.Stop(os.Stdout, os.Stderr)
+		if err != nil {
+			t.Fatalf("failed to stop MySQL: %v", err)
+		}
+	}()
+
+	db, err := sql.Open("mysql", fmt.Sprintf("root@tcp(localhost:%d)/mysql", mysql.Port()))
+	if err != nil {
+		t.Fatalf("failed to open MySQL: %v", err)
+	}
+	defer db.Close()
+
+	_, err = db.Exec("CREATE DATABASE gh_ost_test_db")
+	if err != nil {
+		t.Fatalf("failed to CREATE DATABASE gh_ost_test_db")
+	}
+
+	_, err = db.Exec("USE gh_ost_test_db")
+	if err != nil {
+		t.Fatalf("failed to USE gh_ost_test_db")
+	}
+
+	_, err = db.Exec("CREATE TABLE tbl (id int primary key, data int)")
+	if err != nil {
+		t.Fatalf("failed to CREATE TABLE tbl (id int primary key, data int)")
+	}
+
+	tx, err := db.Begin()
+	if err != nil {
+		t.Fatalf("failed to start a transaction: %v", err)
+	}
+	defer tx.Rollback()
+	for i := 1; i <= 1000; i++ {
```

Comment:
I am not sure we even need 1000, maybe 100 is enough?

Reply:
I think both are ok.